SensorFrame.java

```diff
@@ -0,0 +1,86 @@
+package com.company;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import javax.swing.*;
+import javax.swing.border.TitledBorder;
+
+import static java.awt.BorderLayout.CENTER;
+import static java.awt.BorderLayout.PAGE_END;
+
+public class SensorFrame extends JFrame {
+    public SensorFrame() {
+        setTitle("Sensor Tracker");
+        setLayout(new GridLayout(3,1));
+
+        JPanel  temperaturePnl = new JPanel();
+        temperaturePnl.setBorder(new TitledBorder("Temperature"));
+        add(temperaturePnl);
+        temperaturePnl.setLayout(new BorderLayout());
+
+        SensorInterface temp = new TemperatureAdaptor();

```

Comment:
Because each panel requires the same values and function calls, you could use a data structure to organize making these calls to avoid repeating some of your work three times :)